generators/base/utils.mjs

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright 2013-2022 the original author or authors from the JHipster project.
+ *
+ * This file is part of the JHipster project, see https://www.jhipster.tech/
+ * for more information.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { platform } from 'os';
+import generatorUtils from '../utils.cjs';
+
+const { normalizeLineEndings } = generatorUtils;
+
+const isWin32 = platform() === 'win32';
+
+/**
+ * Converts multiples EditFileCallback callbacks into one.
+ *
+ * @param {...import('./index.cjs').EditFileCallback} callbacks
+ * @returns {import('./index.cjs').EditFileCallback}
+ */
+export const joinCallbacks = (...callbacks) => {
+  return function (content, filePath) {
+    if (isWin32 && content.match(/\r\n/)) {
+      callbacks = [ct => normalizeLineEndings(ct, '\n')].concat(callbacks).concat(ct => normalizeLineEndings(ct, '\r\n'));
+    }
+    for (const callback of callbacks) {
+      content = callback.call(this, content, filePath);
+    }
+    return content;
+  };
+};
+
+export function formatDateForChangelog(now) {
```

Comment:
changelogutils?

Reply:
We will have at least 1 more function there, imo is ok to just call utils.
We can change later when the api become more complete.

File names are not important to apis anymore.
Package.json exports field defines the api now.
We can rearrange at will even after v8 release.

Reply:
I really prefer to have clear file names (strict OOP) that tells what it does compared to surprises